bridge-token-factory/build.sh

```diff
@@ -1,5 +1,17 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+# Exit script as soon as a command fails.
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Comment:
Why not just `DIR="${0%/*}"`?

Reply:
I like @mfornet version better since it uses a tool (`dirname`) that handles path properly while bash incantation may break things out unexpectantly.

Reply:
That line is actually not mine, iirc it is from @ailisp, and is implemented in this way to work well both locally and in CI

Reply:
@frol There are several problems with this approach. First, it creates two new processes just to determine the directory. Second, if `cd` command fails, the script doesn't exit. Third, if the name of the directory ends with a newline, it will be stripped. Also, if it starts with a dash, both `dirname` and `cd` will interpret it as an option, which will also break it.

It is a general problem that shell scripts tend to break when used with unexpected paths or when they encounter unexpected errors. A more robust solution would be:
```sh
if [[ "$0" == "${0%/*}" ]]; then
    DIR=.
else
    DIR="${0%/*}"
fi
```
This should work for all paths, even ones that contain unexpected characters.